### DISCUSSION TITLE
no module named 'freppledb' error !

### DISCUSSION THREAD

**Discussion by Vieirapa** [frePPLe/frepple#637] [Q&A] [answered]
I am trying to following the UBUNTU INSTALATION SCRIPT ( as in https://frepple.com/docs/current/installation-guide/linux-binaries.php#ubuntu-installation-script ) and everything goes Ok until the last step . . .
When I try to run the line 
sudo frepplectl migrate --noinput
I got an error (see the attached image)

![VirtualBox_UBUNTU_27_02_2024_21_15_11](https://github.com/frePPLe/frepple/assets/24809395/bcede1b8-8130-4650-8847-4fc89c4462d9)

Any idea ?

**Comment by jdetaeye** [comment, marked as answer]
Easy and simple to answer.    You're not on a ubuntu 20 instance (which comes with python 3.8 instead of 3.10).
We don't support ubuntu 22 or 23, only unbuntu 20.

FYI: in April ubuntu 24 will be released and we'll have a frepple release 9.0 that jumps from ubuntu 20 to 24 very soon after that.

**Comment by Vieirapa** [comment]
I got it . .. and this was exactly what I made and the 'frepplectl migrate' command runs ok until the end, with no error.
Now we are facing another issue. The SERVER ERROR 500

![ServerError 500](https://github.com/frePPLe/frepple/assets/24809395/c3599563-98d7-4edd-bcc8-73f0ed81be83)

The sequence I worked on was, first to install Apache2 and test (the Welcome page of Apache was displayed) but then, as soon I completed the UBUNTU Script for Frepple install this SERVER ERROR 500 came up!

I am looking at the file  z_frepple.conf at /etc/apache2/sites-available sens to be ok . . but who knows !

Any hint or idea?

**Comment by jdetaeye** [comment]
Looking at apache log file /var/log/apache2/error.log will tell much more about the cause of the error.

**Reply by Vieirapa** [reply]
I did it . . . , here comes the error log file . . . .

[error.log](https://github.com/frePPLe/frepple/files/14469030/error.log)
 
Seems
 to me that is something related to Django, due to the message "AttributeError: module 'Django.db.models' has no attribute 'JSONField'"  . . . I thought at first that maybe it was something related to the Django version

$ django-admin --version
2.2.12

or, do you think this could be an application bug? What am I doing wrong?

best regards, and thanks in advance for your effort to help me!

**Comment by jdetaeye** [comment]
All python packages frepple needs are deployed in a python venv.  It comes packaged with django 4.2.      
That django-admin comes from another application.

Hint: Why not use a docker container avoids all hassle?
